src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1399,6 +1421,12 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                         }
                     }
 
+                    foreach (var hashSlice in _replicationInfo.AttachmentStreamsToDeleteAtTheEndOfBatch)
+                    {
+                        database.DocumentsStorage.AttachmentsStorage.DeleteAttachmentStream(context, hashSlice, expectedCount: 0);
+                        hashSlice.Release(context.Allocator);
```

Comment:
aren't we going to recycle the whole context here?

Reply:
Not sure I follow. I'm releasing the slice?